dspace-spring-rest/src/main/java/org/dspace/app/rest/BitstreamContentRestController.java

```diff
@@ -9,63 +9,141 @@
 
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.core.Response;
 
+import org.apache.log4j.Logger;
 import org.dspace.app.rest.model.BitstreamRest;
-import org.dspace.app.rest.repository.BitstreamRestRepository;
-import org.dspace.core.Utils;
+import org.dspace.app.rest.utils.ContextUtil;
+import org.dspace.app.rest.utils.MultipartFileSender;
+import org.dspace.authorize.AuthorizeException;
+import org.dspace.authorize.service.AuthorizeService;
+import org.dspace.content.Bitstream;
+import org.dspace.content.service.BitstreamService;
+import org.dspace.core.Constants;
+import org.dspace.core.Context;
+import org.dspace.services.ConfigurationService;
+import org.dspace.services.EventService;
+import org.dspace.usage.UsageEvent;
+import org.springframework.beans.factory.InitializingBean;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.hateoas.Link;
-import org.springframework.hateoas.ResourceSupport;
 import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestMethod;
 import org.springframework.web.bind.annotation.RestController;
 
 /**
  * This is a specialized controller to provide access to the bitstream binary content
- * 
- * @author Andrea Bollini (andrea.bollini at 4science.it)
- *
  */
 @RestController
 @RequestMapping("/api/"+BitstreamRest.CATEGORY +"/"+ BitstreamRest.PLURAL_NAME + "/{uuid:[0-9a-fxA-FX]{8}-[0-9a-fxA-FX]{4}-[0-9a-fxA-FX]{4}-[0-9a-fxA-FX]{4}-[0-9a-fxA-FX]{12}}/content")
-public class BitstreamContentRestController {
+public class BitstreamContentRestController implements InitializingBean {
+
+	private static final Logger log = Logger.getLogger(BitstreamContentRestController.class);
+
+	@Autowired
+	private BitstreamService bitstreamService;
+
+	@Autowired
+	private EventService eventService;
+
 	@Autowired
-	private BitstreamRestRepository bitstreamRestRepository; 
-	
-	@RequestMapping(method = RequestMethod.GET)
+	private ConfigurationService configurationService;
+
+	@Autowired
+    private AuthorizeService authorizeService;
+
+	private int bufferSize;
+
+    @Override
+    public void afterPropertiesSet() throws Exception {
+        bufferSize = configurationService.getIntProperty("bitstream-download.buffer.size", -1);
+    }
+
+	@RequestMapping(method = {RequestMethod.GET, RequestMethod.HEAD})
 	public void retrieve(@PathVariable UUID uuid, HttpServletResponse response,
-            HttpServletRequest request) throws IOException {
-		BitstreamRest bit = bitstreamRestRepository.findOne(uuid);
-		if (bit == null) {
-			response.sendError(HttpServletResponse.SC_NOT_FOUND);
-			return;
+											 HttpServletRequest request) throws IOException, SQLException {
+
+		Context context = ContextUtil.obtainContext(request);
+
+        Bitstream bit = getBitstreamIfAuthorized(context, uuid, response);
```

Comment:
We should throw the authorize exception and use the spring mvc excpetion handling here

Reply:
Is it OK if we take the [global exception handling](https://spring.io/blog/2013/11/01/exception-handling-in-spring-mvc#global-exception-handling) approach using `@ControllerAdvice`?